/*
 * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

Since we've forked, folded, mangled, and multilated greatly from jwz's
original, change help links to not point to his docs to avoid user confusion
in the places we've chosen to differ and to avoid our users asking him to
fix things that aren't his fault.

diff -urp -x '*~' driver/XScreenSaver.ad.in driver/XScreenSaver.ad.in
--- driver/XScreenSaver.ad.in	2006-08-08 15:24:27.689831000 -0700
+++ driver/XScreenSaver.ad.in	2006-08-08 15:24:33.346473000 -0700
@@ -85,7 +85,7 @@ GetViewPortIsFullOfLies: False
 
 ! This is the URL loaded by the "Help" button on the splash screen,
 ! and by the "Documentation" menu item in xscreensaver-demo.
-*helpURL: http://www.jwz.org/xscreensaver/man.html
+*helpURL: 
 
 ! loadURL       -- how the "Help" buttons load the helpURL (/bin/sh syntax.)
 ! manualCommand -- how the "Documentation" buttons display man pages.
diff -urp -x '*~' driver/demo-Gtk.c driver/demo-Gtk.c
--- driver/demo-Gtk.c	2006-08-08 15:24:27.699568000 -0700
+++ driver/demo-Gtk.c	2006-08-08 15:24:33.383606000 -0700
@@ -860,6 +860,10 @@ about_menu_cb (GtkMenuItem *menuitem, gp
 G_MODULE_EXPORT void
 doc_menu_cb (GtkMenuItem *menuitem, gpointer user_data)
 {
+
+  const char *sun_help_command = "xterm -T 'Screensaver Help' -e /bin/sh -c 'man xscreensaver || read foo' &";
+
+#ifdef JWZ_HELP /* We do not want to point to jwz website for help instead open man page in xterm window */
   state *s = global_state_kludge;  /* I hate C so much... */
   saver_preferences *p = &s->prefs;
   char *help_command;
@@ -882,6 +886,9 @@ doc_menu_cb (GtkMenuItem *menuitem, gpoi
   if (system (help_command) < 0)
     fprintf (stderr, "%s: fork error\n", blurb());
   free (help_command);
+#endif /*JWZ_HELP*/
+
+  system (sun_help_command);
 }
 
 
diff -urp -x '*~' driver/xscreensaver.c driver/xscreensaver.c
--- driver/xscreensaver.c	2006-05-17 16:35:54.000000000 -0700
+++ driver/xscreensaver.c	2006-08-08 15:24:33.384967000 -0700
@@ -239,9 +239,9 @@ xscreensaver %s, copyright (c) 1991-2006
         xscreensaver &\n\
         xscreensaver-demo\n\
 \n\
-  For updates, online manual, and FAQ, please see the web page:\n\
+  Please read the man page on xscreensaver:\n\
 \n\
-       http://www.jwz.org/xscreensaver/\n\
+       man xscreensaver/\n\
 \n");
 
   fflush (stdout);
@@ -343,9 +343,6 @@ saver_ehandler (Display *dpy, XErrorEven
    "    That won't work.\n");
           fprintf (real_stderr,
    "\n"
-   "    http://www.jwz.org/xscreensaver/bugs.html explains how to create\n"
-   "    the most useful bug reports, and how to examine core files.\n"
-   "\n"
    "    The more information you can provide, the better.  But please\n"
    "    report this bug, regardless!\n"
    "\n");
@@ -412,11 +409,6 @@ startup_ehandler (String name, String ty
            blurb());
     }
 
-  fprintf (stderr, "\n"
-          "              http://www.jwz.org/xscreensaver/faq.html\n"
-          "              http://www.jwz.org/xscreensaver/man.html\n"
-          "\n");
-
   fflush (stderr);
   fflush (stdout);
   exit (1);
@@ -653,8 +645,7 @@ process_command_line (saver_info *si, in
     The `xscreensaver' program is a daemon that runs in the background.\n\
     You control a running xscreensaver process by sending it messages\n\
     with `xscreensaver-demo' or `xscreensaver-command'.\n\
-.   See the man pages for details, or check the web page:\n\
-    http://www.jwz.org/xscreensaver/\n\n");
+.   See the man pages for details.\n\n");
 
 	      /* Since version 1.21 renamed the "-lock" option to "-lock-mode",
 		 suggest that explicitly. */
